padinfo/boardgenerator.py

```diff
@@ -0,0 +1,88 @@
+class BoardGenerator(object):
+    """Generate a Dawnglare link with optional inversion."""
+
+    allowed_letters = "RGBLDHXJMPZ "
+    inversion = "-1"
+    dawnglare_link = "https://pad.dawnglare.com/?patt={}&showfill=1"
+
+    def __init__(self, board):
+        self.board = board.split()
+        self.invalid_board = False
+        self.invert = self.inversion_check()
+        self.generate_clean_board()
+        self.validate_orbs()
+        self.board_length = len(self.clean_board)
+        self.eval_board_size()
+        self.generate_link()
+
+    def inversion_check(self):
+        if BoardGenerator.inversion in self.board:
+            invert = True
+        else:
+            invert = False
+
+        return invert
+
+    def generate_clean_board(self):
+        board = self.board
+
+        if self.invert:
+            board.remove(BoardGenerator.inversion)
+            clean = ''.join(self.invert_board(board))
+        else:
+            clean = ''.join(board)
+
+        self.clean_board = clean
```

Comment:
Even if we're going to define it in a function that \_\_init\_\_ calls, we should always define all of our variables in the \_\_init\_\_ with some default value.  If we know we're definitely going to set it in a subfunction, we can set it to None initially, but it's always good to know exactly what attributes a class has just by looking in the  \_\_init\_\_!